gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsJdbcMonitor.java

```diff
@@ -59,10 +59,15 @@ public ClusterStats monitor(ProxyBackendConfiguration backend) {
     }
     try {
       URL parsedUrl = new URL(url);
+      int urlPort;
+      if (parsedUrl.getPort() != -1) {
+        urlPort = parsedUrl.getPort();
+      } else {
+        urlPort = "https".equals(parsedUrl.getProtocol()) ? 443 : 80;
+      }
```

Comment:
Would `getDefaultPort` be cleaner? https://docs.oracle.com/javase/8/docs/api/java/net/URL.html#getDefaultPort--

Maybe something like `parsedUrl.getPort() == -1 ? parsedUrl.getDefaultPort() : parsedUrl.getPort()`

Will definitely need unit tests for this

Reply:
Oh didn't know about default port. Yes, that sound's much more cleaner.
I tested on my local and it works so i'll ad a unit tests soon.